certora/specs/SocialRecoveryModule.spec

```diff
@@ -436,3 +436,25 @@ rule canFinalizeRecoveryOnlyAfterDelayPeriod(env e) {
 
     assert success => require_uint64(e.block.timestamp) >= recoveryTimestamp, "Recovery finalized before delay period";
 }
+
+// Recovery can be finalized by anyone after the delay period is over.
+rule anyoneCanFinalizeRecovery(env e) {
+    uint64 executeAfter = currentContract.recoveryRequests[safeContract].executeAfter;
+
+    requireInitiatedRecovery(safeContract);
+
+    require e.msg.value == 0;
+    // require require_uint64(e.block.timestamp) >= currentContract.recoveryRequests[safeContract].executeAfter;
+    require safeContract.getOwners().length > 0;
+    // This is to ensure that new owners are not guardians. We cannot use `isGuardian` due to use of quantifiers.
```

Comment:
why do we need to ensure this?

Reply:
Because it checks if the `newOwners[i]` is guardian or not during finalization: https://github.com/5afe/CandideWalletContracts/blob/main/contracts/modules/social_recovery/SocialRecoveryModule.sol#L248

Reply:
And why is it impossible to use `isGuardian` if we remove the quantifier? This reads hard, and I wonder if there are ways to improve it

Reply:
Also, suggestion wrt comment:
```
This is to ensure that new owners are not guardians.
```
doesn't tell me why the LOC is needed. But
```
This contract doesn't allow guardians as new owners, hence we require it.
```
does

Reply:
> why is it impossible to use isGuardian if we remove the quantifier

It is possible to use `isGuardian` without a quantifier.

But we need to make sure that all elements in the `newOwners` address array are not guardians. Not sure how we might be able to do that without a quantifier.

Reply:
Wrote something easier to comprehend using `exists`.